torch/csrc/jit/codegen/cuda/executor_utils.cpp

```diff
@@ -200,6 +200,34 @@ bool validateKernelArgScalar(
   return match;
 }
 
+// Return false if  arg_type doesn't match the type in param
+bool validateKernelArgScalarTensor(
+    const at::Tensor& arg,
+    const Val* param,
+    std::stringstream& msg) {
+  if (!is_cpu_scalar(arg)) {
+    msg << "Argument is a scalar tensor, but the parameter is not."
+        << "\n";
+    return false;
+  }
```

Comment:
1. I'm not planning on doing a `scalar tensor` in fusion IR, since I don't think we want to do anything different for `scalar tensor` vs a normal scalar.
2. Could you also check here that param is either scalar or scalar tensor in that case?
Will do

Reply:
I very much believe we do need to treat it as a tensor for type promotion. I went and piped this through and commited to this branch. Once horizontal fusion of pure CPU scalar tensors is fixed I'm ready to approve and merge. As a follow up @shmsong can you add type promotion tests with cpu scalar tensors?